### PR TITLE
feat: Implement multilingual for site label and description properties - EXO-65344 - Meeds-io/MIPs#72

### PR DESCRIPTION
**Pull request by IlhemEssaadi**

### CODE REVIEW

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

File: webapp/portlet/src/main/webapp/vue-apps/common/js/SiteService.js [hunk 1]
```diff
@@ -6,6 +6,9 @@ export function getSites(siteType, excludedSiteType, excludedSiteName, excludeEm
   if (excludedSiteType) {
     formData.append('excludedSiteType', excludedSiteType);
   }
+  
+  formData.append('lang', eXo.env.portal.language);
+
   if (excludedSiteName) {
     formData.append('excludedSiteName', excludedSiteName);
   }
```
Comment (azayati) [inline webapp/portlet/src/main/webapp/vue-apps/common/js/SiteService.js]:
```suggestion
  if (excludedSiteName) {
    formData.append('excludedSiteName', excludedSiteName);
  }  
  formData.append('lang', eXo.env.portal.language);
```

File: extension/war/src/main/webapp/WEB-INF/conf/social-extension/social/core-configuration.xml [hunk 2]
```diff
@@ -879,4 +879,13 @@
...
     </external-component-plugins>
 
+    <external-component-plugins>
+        <target-component>io.meeds.social.translation.service.TranslationService</target-component>
+        <component-plugin>
+            <name>site</name>
+            <set-method>addPlugin</set-method>
+            <type>org.exoplatform.social.core.plugin.SiteTranslationPlugin</type>
+        </component-plugin>
+    </external-component-plugins>
```
Comment (azayati) [inline extension/war/src/main/webapp/WEB-INF/conf/social-extension/social/core-configuration.xml]:
```suggestion
    <external-component-plugins>
      <target-component>io.meeds.social.translation.service.TranslationService</target-component>
      <component-plugin>
         <name>site</name>
          <set-method>addPlugin</set-method>
          <type>org.exoplatform.social.core.plugin.SiteTranslationPlugin</type>
      </component-plugin>
    </external-component-plugins>
```

File: component/service/src/main/java/org/exoplatform/social/rest/impl/site/SiteRest.java [hunk 3]
```diff
@@ -109,7 +110,10 @@ public Response getSites(@Context
                            @Parameter(description = "Limit of results to retrieve")
                            @QueryParam("limit")
                            @DefaultValue("0")
-                           int limit) {
+                           int limit,
+                           @Parameter(description = "Used to retrieve the title and description in requested language")
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/site/SiteRest.java]:
```suggestion
                           @Parameter(description = "Used to retrieve the site label and description in the requested language")
```

File: component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java [hunk 4]
```diff
@@ -18,6 +18,9 @@
 package org.exoplatform.social.rest.api;
 
 import static org.exoplatform.portal.mop.rest.EntityBuilder.toUserNodeRestEntity;
+import static org.exoplatform.social.core.plugin.SiteTranslationPlugin.SITE_OBJECT_TYPE;
+import static org.exoplatform.social.core.plugin.SiteTranslationPlugin.SITE_TITLE_FIELD_NAME;
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java]:
```suggestion
import static org.exoplatform.social.core.plugin.SiteTranslationPlugin.SITE_LABEL_FIELD_NAME;
```

File: component/core/src/main/java/org/exoplatform/social/core/plugin/SiteTranslationPlugin.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+import org.gatein.api.site.Site;
+import org.gatein.api.site.SiteId;
+
+public class SiteTranslationPlugin extends TranslationPlugin {
+
+  public static final String SITE_OBJECT_TYPE            = "site";
+
+  public static final String SITE_DESCRIPTION_FIELD_NAME = "description";
+
+  public static final String SITE_TITLE_FIELD_NAME       = "title";
```
Comment (azayati) [inline component/core/src/main/java/org/exoplatform/social/core/plugin/SiteTranslationPlugin.java]:
```suggestion
  public static final String SITE_LABEL_FIELD_NAME       = "label";
```

File: component/core/src/main/java/org/exoplatform/social/core/plugin/SiteTranslationPlugin.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  public String getObjectType() {
+    return SITE_OBJECT_TYPE;
+  }
+
+  @Override
+  public boolean hasAccessPermission(long siteId, String username) throws ObjectNotFoundException {
+    PortalConfig portalConfig = layoutService.getPortalConfig(siteId);
+    SiteId id = Util.from(new SiteKey(portalConfig.getType(), portalConfig.getName()));
+    Site site = portal.getSite(id);
+    UserACL userACL = CommonsUtils.getService(UserACL.class);
```
Comment (azayati) [inline component/core/src/main/java/org/exoplatform/social/core/plugin/SiteTranslationPlugin.java]: inject UserAcl service in the constructor instead

File: component/core/src/main/java/org/exoplatform/social/core/plugin/SiteTranslationPlugin.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    return userACL.hasPermission(Util.from(site.getEditPermission())[0]);
+  }
+
+  @Override
+  public boolean hasEditPermission(long siteId, String username) throws ObjectNotFoundException {
+    PortalConfig portalConfig = layoutService.getPortalConfig(siteId);
+    SiteId id = Util.from(new SiteKey(portalConfig.getType(), portalConfig.getName()));
+    Site site = portal.getSite(id);
+    UserACL userACL = CommonsUtils.getService(UserACL.class);
+    return userACL.hasPermission(Util.from(site.getAccessPermission())[0]);
```
Comment (azayati) [inline component/core/src/main/java/org/exoplatform/social/core/plugin/SiteTranslationPlugin.java]:
```suggestion
    return userACL.hasEditPermission(portalConfig);
```

File: component/core/src/main/java/org/exoplatform/social/core/plugin/SiteTranslationPlugin.java [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+  @Override
+  public String getObjectType() {
+    return SITE_OBJECT_TYPE;
+  }
+
+  @Override
+  public boolean hasAccessPermission(long siteId, String username) throws ObjectNotFoundException {
+    PortalConfig portalConfig = layoutService.getPortalConfig(siteId);
+    SiteId id = Util.from(new SiteKey(portalConfig.getType(), portalConfig.getName()));
+    Site site = portal.getSite(id);
```
Comment (azayati) [inline component/core/src/main/java/org/exoplatform/social/core/plugin/SiteTranslationPlugin.java]:
```suggestion
    Site site = new SiteImpl(portalConfig)
```

File: component/core/src/main/java/org/exoplatform/social/core/plugin/SiteTranslationPlugin.java [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    UserACL userACL = CommonsUtils.getService(UserACL.class);
+    return userACL.hasPermission(Util.from(site.getEditPermission())[0]);
```
Comment (azayati) [inline component/core/src/main/java/org/exoplatform/social/core/plugin/SiteTranslationPlugin.java]: add a new method hasAccessPermission(portalConfig) for userAcl and use it directly here

File: component/core/src/main/java/org/exoplatform/social/core/plugin/SiteTranslationPlugin.java (hunk 8 above)
Comment (azayati) [reply component/core/src/main/java/org/exoplatform/social/core/plugin/SiteTranslationPlugin.java]: not needed if we will use the new method userAcl.hasAccessPermission(portalConfig)

File: component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java [hunk 10]
```diff
@@ -1825,10 +1834,16 @@ public static SiteEntity buildSiteEntity(PortalConfig site,
                                              getLayoutService(),
                                              getUserACL());
     }
-    return new SiteEntity(siteType,
+    PortalConfig sitePortalConfig = getLayoutService().getPortalConfig(new SiteKey(siteType, site.getName()));
+    long siteId = Long.parseLong((sitePortalConfig.getStorageId().split("_"))[1]);
+    String translatedTitle = getTranslatedLabel(SITE_TITLE_FIELD_NAME, siteId, locale);
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java]:
```suggestion
    String translatedSiteLabel = getTranslatedLabel(SITE_LABEL_FIELD_NAME, siteId, locale);
```

File: component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java [hunk 11]
```diff
@@ -1825,10 +1834,16 @@ public static SiteEntity buildSiteEntity(PortalConfig site,
...
+    String translatedDescription= getTranslatedLabel(SITE_DESCRIPTION_FIELD_NAME, siteId, locale);
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java]:
```suggestion
    String translatedSiteDescription= getTranslatedLabel(SITE_DESCRIPTION_FIELD_NAME, siteId, locale);
```

File: component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java [hunk 12]
```diff
@@ -1825,10 +1834,16 @@ public static SiteEntity buildSiteEntity(PortalConfig site,
...
+    displayName = StringUtils.isNotBlank(translatedTitle) ? translatedTitle : displayName;
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/api/EntityBuilder.java]:
```suggestion
    String displayName = StringUtils.isNotBlank(translatedTitle) ? translatedTitle : site.getLabel();
```
Remove the first instruction String displayName = site.getLabel(); above

File: component/service/src/main/java/org/exoplatform/social/rest/entity/SiteEntity.java [hunk 13]
```diff
@@ -31,6 +31,8 @@
 @NoArgsConstructor
 public class SiteEntity {
 
+  private long                      StorageId;
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/entity/SiteEntity.java]: siteId

**Review by azayati — CHANGES_REQUESTED** [review]

File: component/service/src/main/java/org/exoplatform/social/rest/impl/site/SiteRest.java [hunk 14]
```diff
@@ -130,10 +134,14 @@ public Response getSites(@Context
...
-      return Response.ok(EntityBuilder.buildSiteEntities(sites, request, expandNavigations, excludeEmptyNavigationSites, filterByPermission, sortByDisplayOrder)).build();
+      return Response.ok(EntityBuilder.buildSiteEntities(sites, request, expandNavigations, excludeEmptyNavigationSites, filterByPermission, sortByDisplayOrder, getLocale(lang))).build();
     } catch (Exception e) {
       LOG.warn("Error while retrieving sites", e);
       return Response.serverError().build();
     }
   }
+
+  private Locale getLocale(String lang) {
+    return org.apache.commons.lang3.StringUtils.isBlank(lang) ? null : Locale.forLanguageTag(lang);
```
Comment (azayati) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/site/SiteRest.java]: use org.apache.commons.lang.StringUtils instead

**Review by azayati — CHANGES_REQUESTED** [review]

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**